packages/hub/test/hub.test.ts

```diff
@@ -139,6 +139,19 @@ describe('Hub', () => {
       });
       expect(hub.getStack()).toHaveLength(1);
     });
+
+    test('returns return value from wrapped function', () => {
+      // someFn represents an existing function
+      const someFn = () => {
+        const hub = getCurrentHub();
+        hub.setTag('key', 'value');
+        hub.captureMessage('test');
+        return 'ok';
+      };
+      const hub = new Hub();
+      const value = hub.withScope(someFn); // wraps someFn to run it in a new scope
```

Comment:
I wanted to write...

```suggestion
      // wrapped wraps someFn to run it in a new scope
      const wrapped = () => hub.withScope(someFn);
      const value = wrapped();
```

... because we wanted `withScope` (and, eventually, `Sentry.trace`) to be easier to write wrapped versions of existing functions.

However, by delaying evaluation with `() => ...`, we're also delaying the evaluation of what is the current scope, which is not what one would typically want. Thus, the current form is only good for wrap-and-call existing functions.